app/controllers/profiles_controller.rb

```diff
@@ -0,0 +1,36 @@
+class ProfilesController < ApplicationController
+  before_action :set_profile, only: [:show, :edit, :update, :destroy]
+
+  def set_current_deck
+    unless current_user.set_deck(deck_params)
+      flash[:deck_message] = "Something wrong happened."
+    end
+    redirect_to decks_path
+  end
+
+  def edit
+  end
+
+  def update
+    if @profile.update(profile_params)
+      redirect_to @profile, notice: "User was successfully updated."
+    else
+      render action: 'edit'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.